pkg/controller/infrastructure/templates/main.tpl.tf

```
@@ -4,9 +4,9 @@ provider "packet" {
 
 // Deploy a new ssh key
 resource "packet_project_ssh_key" "publickey" {
-  name = "{{ required "clusterName is required" .Values.clusterName }}-ssh-publickey"
-  public_key = "{{ required "sshPublicKey is required" .Values.sshPublicKey }}"
-  project_id = "{{ required "packet.projectID is required" .Values.packet.projectID }}"
+  name = "{{ .Values.clusterName }}-ssh-publickey"
+  public_key = "{{ .Values.sshPublicKey }}"
+  project_id = "{{ .Values.packet.projectID }}"
```

```suggestion
  name = "{{ .clusterName }}-ssh-publickey"
  public_key = "{{ .sshPublicKey }}"
  project_id = "{{ .packet.projectID }}"
```

Right? Same would apply in line 16.

---

Nice catch... :) thanks!